backend/auth/handlers.go

```diff
@@ -0,0 +1,340 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Canonical Ltd.
+
+package auth
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"math/big"
+	mrand "math/rand"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/omec-project/webconsole/backend/logger"
+	"github.com/omec-project/webconsole/configmodels"
+	"github.com/omec-project/webconsole/dbadapter"
+	"go.mongodb.org/mongo-driver/bson"
+	"golang.org/x/crypto/bcrypt"
+)
+
+const (
+	errorCreateUserAccount    = "failed to create user account"
+	errorDeleteAdminAccount   = "deleting an Admin account is not allowed"
+	errorDeleteUserAccount    = "failed to delete user account"
+	errorIncorrectCredentials = "incorrect username or password. Try again"
+	errorInvalidDataProvided  = "invalid data provided"
+	errorInvalidPassword      = "Password must have 8 or more characters, must include at least one capital letter, one lowercase letter, and either a number or a symbol."
+	errorLogin                = "failed to log in"
+	errorMissingPassword      = "password is required"
+	errorMissingUsername      = "username is required"
+	errorRetrieveUserAccount  = "failed to retrieve user account from DB"
+	errorRetrieveUserAccounts = "failed to retrieve user accounts from DB"
+	errorUpdateUserAccount    = "failed to update user account"
+	errorUsernameNotFound     = "username not found"
+	UserAccountDataColl       = "webconsoleData.snapshots.userAccountData"
+)
+
+func GetUserAccounts(c *gin.Context) {
+	users, err := fetchUsers()
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	c.JSON(http.StatusOK, users)
+}
+
+func fetchUsers() ([]*configmodels.User, error) {
+	rawUsers, err := dbadapter.WebuiDBClient.RestfulAPIGetMany(UserAccountDataColl, bson.M{})
+	if err != nil {
+		logger.DbLog.Errorln(err.Error())
+		return nil, errors.New(errorRetrieveUserAccounts)
+	}
+	var users []*configmodels.User
+	users = make([]*configmodels.User, 0)
+	for _, rawUser := range rawUsers {
+		var user configmodels.User
+		err := json.Unmarshal(configmodels.MapToByte(rawUser), &user)
+		if err != nil {
+			logger.DbLog.Errorf(errorRetrieveUserAccount)
+			continue
+		}
+		user.Password = ""
+		users = append(users, &user)
+	}
+	return users, nil
+}
+
+func GetUserAccount(c *gin.Context) {
+	logger.WebUILog.Infoln("get user account")
+	username := c.Param("username")
+	user, err := fetchUser(username)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorRetrieveUserAccount})
+		return
+	}
+	if user == nil {
+		c.JSON(http.StatusNotFound, gin.H{"error": errorUsernameNotFound})
+		return
+	}
+	user.Password = ""
```

Comment:
This is not great, the object we return should be different from the one in the DB. 

```go
type GetUserResponse struct {
	Username    string `json:"username"`
	Permissions int    `json:"permissions"`
}
```